tests/test_inner.py

```diff
@@ -526,6 +609,19 @@ class Exit(ExitPlugin):
      False,  # does not log error
     ),
 
+    # No such plugin, buildstep, not required
+    ({'buildstep_plugins': [{'name': 'no plugin',
+                             'args': {},
+                             'required': False},
+                            {'name': 'buildstep_watched',
+                             'args': {
+                                 'watcher': Watcher(),
+                             }
+                            }]},
+     False,  # not fatal
+     False,  # does not log error
+    ),
+
```

Comment:
Can we add a test here to make sure the build doesn't get marked as failed in that situation?